### DISCUSSION TITLE
All Caps Person entity

### DISCUSSION THREAD

**Discussion by storkeu** [microsoft/presidio#908] [Q&A]
Hello!

In my text, if a person is in ALL CAPS form, presidio is unable to recogize as PERSON.... same if all lower but I think that is understandable

Reasons? Solutions?

Thank you

**Comment by omri374** [comment]
Hi @storkeu. Presidio uses a default NER model from spaCy (`en_core_web_lg`). However this is just an example implementation. I would suggest to look into other types of NER models which were trained on data containing all caps person names, or training your own. An alternative (depending on the case) is to use a rule-based recognizer comprised of a list of names in capital letters. 

See these tutorials:
1. Rule based (no-code) recognizers: https://microsoft.github.io/presidio/tutorial/08_no_code/
2. Adapting the NER engine: 
  - [Using a different spaCy model](https://microsoft.github.io/presidio/tutorial/05_languages/)
  - [Using flair based NER models](https://github.com/microsoft/presidio/blob/main/docs/samples/python/flair_recognizer.py)
  - [Using huggingface based NER models](https://github.com/microsoft/presidio/blob/main/docs/samples/python/transformers_recognizer/index.md)
